lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -85,18 +75,39 @@ final class ViewFocusBinding {
         direction: _viewFocusDirection,
       );
     }
+    _markViewAsFocusable(_lastViewId, reachableByKeyboard: true);
+    _markViewAsFocusable(viewId, reachableByKeyboard: false);
     _lastViewId = viewId;
-    _notify(event);
+    _onViewFocusChange(event);
   }
 
-  static int? _viewId(DomElement? element) {
-    final DomElement? viewElement = element?.closest(
-      DomManager.flutterViewTagName,
-    );
-    final String? viewIdAttribute = viewElement?.getAttribute(
-      GlobalHtmlAttributes.flutterViewIdAttributeName,
-    );
-    return viewIdAttribute == null ? null : int.tryParse(viewIdAttribute);
+  int? _viewId(DomElement? element) {
```

Comment:
Basically it'd guarantee in this case that every `element` would have a parent (or itself) with a flutter view id, so we can make this an `int _viewId` function, no?